controllers/observability/observability_controller.go

```diff
@@ -0,0 +1,71 @@
+package observability
+
+import (
+	"context"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/alertmanager"
+)
+
+var (
+	log         = logf.Log.WithName("controller_observability")
+	periodicity = 1 * time.Hour
+)
+
+type Reconciler struct {
+	config *rest.Config
+	events chan event.GenericEvent
+
+	amApi *alertmanager.Api
+}
+
+func (r *Reconciler) Reconcile(_ context.Context, _ ctrl.Request) (ctrl.Result, error) {
+	log.Info("Reconciling Observability")
+
+	if err := r.ensurePodDisruptionBudgetAtLimitIsSilenced(); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func NewReconciler(config *rest.Config) *Reconciler {
+	return &Reconciler{
+		config: config,
+		events: make(chan event.GenericEvent),
```

Comment:
I think it will be safer to have a size-1 cached channel:
```suggestion
		events: make(chan event.GenericEvent, 1),
```
That should prevent crasy, hard-to-debug dead loops

Reply:
makes sense, added